lib/tcp.js

```diff
@@ -0,0 +1,166 @@
+/* Copyright (c) 2013-2015 Richard Rodger, MIT License */
+/* jshint node:true, asi:true, eqnull:true */
+'use strict'
+
+// Load modules
+var Util = require('util')
+var Net = require('net')
+var Stream = require('stream')
+var Ndjson = require('ndjson')
+var Reconnect = require('reconnect-core')
+
+// Declare internals
+var internals = {}
```

Comment:
I love it. I want to talk to @mcollina about perf impact here because I want to use this on all my libs. Makes reading the code super nice.